app/models/manager_refresh/save_collection/helper.rb

```diff
@@ -50,9 +50,16 @@ def create_or_update_inventory!(inventory_collection, record_index, association)
       unique_index_keys = inventory_collection.manager_ref_to_cols
 
       association.find_each do |record|
-        # TODO(lsmola) the old code was able to deal with duplicate records, should we do that? The old data still can
-        # have duplicate methods, so we should clean them up. It will slow up the indexing though.
-        record_index[inventory_collection.object_index_with_keys(unique_index_keys, record)] = record
+        index = inventory_collection.object_index_with_keys(unique_index_keys, record)
```

Comment:
ah, ok, in that case if would be just a matter of changing unique index for the vmware, which is easy in graph refresh

and VmWare would still be specific with looking at Vm, not ems.vms. Since it can bring Vms back to life, by doing a cross region migration, right? I will need to explore this in the clouds. But I would like to avoid looking at all Mms, so maybe this could be somehow wraped by a Provider model.